src/core/installAddon.js

```diff
@@ -333,6 +334,29 @@ export class WithInstallHelpers extends React.Component {
     }
   }
 
+  getCurrentStatus() {
+    const {
+      _addonManager,
+      addon: { guid },
+      hasAddonManager,
+    } = this.props;
+
+    if (!hasAddonManager) {
+      log.info('No addon manager, cannot set add-on status');
+      return Promise.resolve();
+    }
+
+    log.info('Getting add-on status');
+    return _addonManager
+      .getAddon(guid)
+      .then((addon) => {
+        return addon.isEnabled;
```

Comment:
I see that this code is the same as `setCurrentStatus`, which is called on every prop change.

I did not know we were using `getAddon` before so I wonder if this is needed. Maybe relying on `status` is enough.

Reply:
Ok, I missed the condition on the GUID. That makes sense performance-wise.

> I don't know that if we'll we see ENABLED right after install. I compared by testing install of a regular extension and its status right after install shows UNINSTALLED...which doesn't seem right to me..? So there may be a deeper issue here to fix.

oO, that does not sound right.

>  For now maybe I can check for the DISABLED state (in case it gets fixed and is working more like an extension and show UNINSTALLED). if that all makes sense : )

We can probably live with a new method in `installAddon`. I'd like to investigate more the potential bug above.

Reply:
More on this

> I don't know that if we'll we see ENABLED right after install. I compared by testing install of a regular extension and its status right after install shows UNINSTALLED...which doesn't seem right to me..? So there may be a deeper issue here to fix.

In the render, it does show a more accurate status. For a regular extension it says INSTALLED. If you refresh the page it shows ENABLED.

I think after the install, its just not updated at that point yet so it actually still shows the older status..